docs/backend/behaviors.md

```diff
@@ -11,12 +11,305 @@ myst:
+
+You create a schema with the net value for the form and attributes for the calculated values.
+You create an adapter to calculate the VAT and gross values.
+You need a marker interface to distinguish between context and adapter.
+
+The Python code in a file `price.py` looks like so:
+
+```python
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.supermodel import model
+from zope import schema
+from zope.interface import Attribute
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.interface import provider
+
+
+@provider(IFormFieldProvider)
+class IPriceBehavior(model.Schema):
+    """Behavior: a price, VAT and gross."""
+
+    price_net = schema.Float(
+        title="Price (net)",
+        required=True,
+    )
+    price_vat = Attribute("VAT 20% of net price")
+    price_gross = Attribute("Price gross (net + VAT 20%")
+
+
+class IPriceMarker(Interface):
+    """Marker for content that has a price."""
+
+
+@implementer(IPriceBehavior)
+class PriceAdapter:
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def price_net(self):
+        """Getter, read from context and return back"""
+        return self.context.price_net
+
+    @price_net.setter
+    def price_net(self, value):
+        """Setter, called by the form, set on context"""
+        self.context.price_net = value
+
+    @property
+    def price_vat(self):
+        return self.price_net * 0.2
+
+    @property
+    def price_gross(self):
+        return self.price_net + self.price_vat
+
+```
+
+The registration in the `configure.zcml` looks like so:
+
+```XML
+  <plone:behavior
+      factory=".price.PriceAdapter"
+      for=".price.IPriceMarker"
+      marker=".price.IPriceMarker"
+      name="myproject.price"
+      provides=".price.IPriceBehavior"
+      title="Price with net, VAT and gross"
+  />
+  ```
+
+After a restart of Plone, the behavior can be added to the type in the Content Types control panel.
+The add and edit forms are containing a new field `Price (net)`.
+
+This field is not displayed in most views.
+To display the data entered in this field you need to modify the page template and access the `price_net` field as `context.price_net` there.
+To access the `price_vat` and `price_gross` fields you need to get the adapter in your view class like so:
+
+```Python
+from .price import IPriceBehavior
+
+class SomeViewClass:
+
+    def vat(self):
+        adapter = IPriceBehavior(context)
+        return adapter.price_vat
+
+    def gross(self):
+        adapter = IPriceBehavior(context)
+        return adapter.price_gross
+```
+
+### Creating a behavior with PloneCLI
+
+To add a behavior to your add-on, you can use PloneCLI as follows:
+
+```bash
+plonecli add behavior
+```
+
+This will create the behavior Python file in the `behaviors` folder where you can define your behaviors schema fields and registers the behavior in the `configure.zcml`.
```

Comment:
```suggestion
This will create the behavior Python file in the `behaviors` folder, where you can define your behaviors schema fields, and registers the behavior in the `configure.zcml`.
```